test/integration/integration_test.go

```diff
@@ -442,7 +443,21 @@ func exportKindLogs(t *testing.T) {
 	}
 }
 
-func assertRolloutRestarts(t *testing.T, ctx context.Context, client ctrlclient.Client, obj ctrlclient.Object, targets []secretsv1alpha1.RolloutRestartTarget) {
+func waitForRolloutRestartsAndAssertRollout(t *testing.T, ctx context.Context,
+	client ctrlclient.Client, obj ctrlclient.Object, targets []secretsv1alpha1.RolloutRestartTarget,
+) {
+	t.Helper()
+	require.NoError(t, backoff.Retry(
+		func() error {
+			return assertRolloutRestarts(t, ctx, client, obj, targets)
```

Comment:
Suggest handling the case where the `assertRolloutRestarts` failed the test. Break out of the retry early.
```suggestion
			err := assertRolloutRestarts(t, ctx, client, obj, targets)
			if t.Failed() {
				e := fmt.Errorf("assertRolloutRestarts failed")
				if err != nil {
					e = fmt.Errorf("%s, err=%w", e.Error(), err)
				}
				return backoff.Permanent(e)
			}
			return err
```